Helper/Api/Products.php

```diff
@@ -771,24 +776,21 @@ public function update($storeId, $target = null, array $result = [])
             }
 
             $sourceItemData = [
-                'source_code' => 'default',
+                'source_code' => $this->configHelper->getStockSource(),
                 'status' => 1
             ];
 
             $shouldUpdateStock = false;
             $stockItem = $this->stockItem->load($this->productRepository->get($sku)->getId(), 'product_id');
 
-            (float) $product_stock_value = $product['product_stock']['value'];
+            (float) $product_stock_value = (array_key_exists('orderable_stock_value', $result)) ? $result['orderable_stock_value'] : null;
```

Comment:
```suggestion
            $product_stock_value = (array_key_exists('orderable_stock_value', $result)) ? $result['orderable_stock_value'] : null;
```

float here does not do anything